homeassistant/components/tolo/sensor.py

```diff
@@ -0,0 +1,73 @@
+"""TOLO Sauna (non-binary, general) sensors."""
+
+from homeassistant.components.sensor import SensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ENTITY_CATEGORY_DIAGNOSTIC, PERCENTAGE, TEMP_CELSIUS
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ToloSaunaCoordinatorEntity, ToloSaunaUpdateCoordinator
+from .const import DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up (non-binary, general) sensors for TOLO Sauna."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        [
+            ToloWaterLevelSensor(coordinator, entry),
+            ToloTankTemperatureSensor(coordinator, entry),
+        ]
+    )
+
+
+class ToloWaterLevelSensor(ToloSaunaCoordinatorEntity, SensorEntity):
+    """Sensor for tank water level."""
+
+    WATER_LEVELS = {0: 0, 1: 33, 2: 66, 3: 100}
+
+    _attr_entity_category = ENTITY_CATEGORY_DIAGNOSTIC
+    _attr_name = "Water Level"
+    _attr_icon = "mdi:waves-arrow-up"
+    _attr_state_class = "measurement"
+    _attr_native_unit_of_measurement = PERCENTAGE
+
+    def __init__(
+        self, coordinator: ToloSaunaUpdateCoordinator, entry: ConfigEntry
+    ) -> None:
+        """Initialize TOLO Sauna tank water level sensor entity."""
+        super().__init__(coordinator, entry)
+
+        self._attr_unique_id = f"{entry.entry_id}_water_level"
+
+    @property
+    def native_value(self) -> int:
+        """Return current tank water level."""
+        return self.WATER_LEVELS[self.coordinator.data.status.water_level]
+
+
+class ToloTankTemperatureSensor(ToloSaunaCoordinatorEntity, SensorEntity):
+    """Sensor for tank temperature."""
+
+    _attr_entity_category = ENTITY_CATEGORY_DIAGNOSTIC
+    _attr_name = "Tank Temperature"
+    _attr_icon = "mdi:thermometer"
```

Comment:
As this is a temperature, we can set the device class for it instead.
```suggestion
    _attr_device_class = DEVICE_CLASS_TEMPERATURE
```